--- ./patchsource/www/rega/esp/channels.fn.orig
+++ ./patchsource/www/rega/esp/channels.fn
@@ -752,6 +752,38 @@
   {
     object oOL = chn.DPByControl("DIMMER.OLD_LEVEL");
     object oL = chn.DPByControl("DIMMER.LEVEL");
+    if( oL && oOL )
+    {
+      oOL.State(1);
+    }
+  }
+  return;
+}
+
+! [chnId]
+function ::servoOldVal()
+{
+  object chn = dom.GetObject( system.GetVar("chnId") );
+  if (chn)
+  {
+    object oOL = chn.DPByControl("SERVO.OLD_LEVEL");
+    object oL = chn.DPByControl("SERVO.LEVEL");
+    if( oL && oOL )
+    {
+      oOL.State(1);
+    }
+  }
+  return;
+}
+
+! [chnId]
+function ::fanOldVal()
+{
+  object chn = dom.GetObject( system.GetVar("chnId") );
+  if (chn)
+  {
+    object oOL = chn.DPByControl("FAN.OLD_LEVEL");
+    object oL = chn.DPByControl("FAN.LEVEL");
     if( oL && oOL )
     {
       oOL.State(1);
@@ -761,6 +793,22 @@
 }
 
 ! [chnId]
+function ::airflapOldVal()
+{
+  object chn = dom.GetObject( system.GetVar("chnId") );
+  if (chn)
+  {
+    object oOL = chn.DPByControl("AIRFLAP.OLD_LEVEL");
+    object oL = chn.DPByControl("AIRFLAP.LEVEL");
+    if( oL && oOL )
+    {
+      oOL.State(1);
+    }
+  }
+  return;
+}
+
+! [chnId]
 ! --> bhasReadDPs
 function ::chnHasReadDPs() {
   boolean bHasReadDPs = false;
